pkg/apiserver/resthandler.go

```diff
@@ -234,7 +256,7 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
 		opts := *out.(*api.ListOptions)
```

Comment:
Can you add a TODO above this line? This will prevent different groups/resources from having their own ListOptions type. We need to switch to something like requiring ListOptions types to implement an interface, or explicitly making an api(?).ListOptions and running conversion on it (requiring users to provide the appropriate conversion functions if necessary).